tests/common.rs

```diff
@@ -184,17 +209,333 @@ pub fn run_native_program(
+                    let native_val: bool = native_tag == 0; // 0 = true
+                    prop_assert_eq!(vn_val, native_val, "bool value mismatch");
+                } else if is_panic {
+                    check_next_type(
+                        reg.get_type(&info.variants[native_tag as usize]).unwrap(),
+                        ignore_gas,
+                        &mut [&enum_container[1]].into_iter(),
+                        vm_rets,
+                        vm_gas,
+                        reg,
+                    )?;
+                } else {
+                    let vm_tag = vm_rets.next().unwrap();
+                    let vm_tag = casm_variant_to_sierra(
+                        vm_tag.parse::<i64>().unwrap(),
+                        info.variants.len() as i64,
+                    ) as u64;
+                    prop_assert_eq!(vm_tag, native_tag, "enum tag mismatch");
+
+                    check_next_type(
+                        reg.get_type(&info.variants[native_tag as usize]).unwrap(),
+                        ignore_gas,
+                        &mut [&enum_container[1]].into_iter(),
+                        vm_rets,
+                        vm_gas,
+                        reg,
+                    )?;
+                }
+            }
+            CoreTypeConcrete::Struct(info) => {
+                let struct_container = native_rets.next().unwrap().as_array().unwrap();
+                for (field, container) in info.members.iter().zip(struct_container.iter()) {
+                    check_next_type(
+                        reg.get_type(field).unwrap(),
+                        ignore_gas,
+                        &mut [container].into_iter(),
+                        vm_rets,
+                        vm_gas,
+                        reg,
+                    )?;
+                }
+            }
+            CoreTypeConcrete::Felt252Dict(_) => todo!(),
+            CoreTypeConcrete::Felt252DictEntry(_) => todo!(),
+            CoreTypeConcrete::SquashedFelt252Dict(_) => todo!(),
+            CoreTypeConcrete::Pedersen(_) => {
+                // runner: ignore
+                // native: null
+                native_rets
+                    .next()
+                    .unwrap()
+                    .as_null()
+                    .expect("should be null");
+            }
+            CoreTypeConcrete::Poseidon(_) => todo!(),
+            CoreTypeConcrete::Span(_) => todo!(),
+            CoreTypeConcrete::StarkNet(_) => todo!(),
+            CoreTypeConcrete::SegmentArena(_) => todo!(),
+            CoreTypeConcrete::Snapshot(_) => todo!(),
+        }
+
+        Ok(())
+    }
+
+    for ty in ret_types {
+        let ty = reg.get_type(ty).unwrap();
+        check_next_type(ty, ignore_gas, &mut native_rets, &mut vm_rets, vm_gas, &reg)?;
+    }
+
+    Ok(())
+}
+
+pub const FIELD_HIGH: u128 = (1 << 123) + (17 << 64); // this is equal to 10633823966279327296825105735305134080
+pub const FIELD_LOW: u128 = 1;
```

Comment:
I think this is `felt252`'s prime modulo, which we have available at `cairo_native::types::felt252::PRIME` if I'm not mistaken.

Reply:
i guess, but the high and low fields are used in the proptest value generation

i mainly copied it from the cairo_felt crate, which has a feature for this, but because cairo-vm uses an older version which doesn't have the feature, i had to copy this over.